client/src/main/java/io/pravega/client/connection/impl/AppendBatchSizeTrackerImpl.java

```diff
@@ -27,19 +30,31 @@
  * synchronous writers. Otherwise the batch size is set to the amount of data that will be written in the next
  * {@link #MAX_BATCH_TIME_MILLIS} or half the server round trip time (whichever is less)
  */
-class AppendBatchSizeTrackerImpl implements AppendBatchSizeTracker {
-    private static final int MAX_BATCH_TIME_MILLIS = 100;
-
+public class AppendBatchSizeTrackerImpl implements AppendBatchSizeTracker {
+    
+    // This must be less than WireCommands.MAX_WIRECOMMAND_SIZE / 2;
+    @VisibleForTesting
+    static final int MAX_BATCH_SIZE = EnvVars.readIntegerFromEnvVar("PRAVEGA_MAX_BATCH_SIZE",
+                                                                 2 * TcpClientConnection.TCP_BUFFER_SIZE - 1024);
+    @VisibleForTesting
+    static final int BASE_TIME_NANOS = EnvVars.readIntegerFromEnvVar("PRAVEGA_BATCH_BASE_TIME_NANOS", AbstractTimer.NANOS_TO_MILLIS / 2);
+    @VisibleForTesting
+    static final int BASE_SIZE = EnvVars.readIntegerFromEnvVar("PRAVEGA_BATCH_BASE_SIZE", 0);
+    @VisibleForTesting
+    static final double OUTSTANDING_FRACTION = 1.0 / EnvVars.readIntegerFromEnvVar("PRAVEGA_BATCH_OUTSTANDING_DENOMINATOR", 2);
+    
+    private static final double NANOS_PER_MILLI = AbstractTimer.NANOS_TO_MILLIS;
+    
     private final Supplier<Long> clock;
     private final AtomicLong lastAppendNumber;
     private final AtomicLong lastAppendTime;
     private final AtomicLong lastAckNumber;
-    private final ExponentialMovingAverage eventSize = new ExponentialMovingAverage(1024, 0.1, true);
-    private final ExponentialMovingAverage millisBetweenAppends = new ExponentialMovingAverage(10, 0.1, false);
-    private final ExponentialMovingAverage appendsOutstanding = new ExponentialMovingAverage(2, 0.05, false);
+    private final ExponentialMovingAverage eventSize = new ExponentialMovingAverage(1024, 0.01, true);
+    private final ExponentialMovingAverage nanosBetweenAppends = new ExponentialMovingAverage(10 * NANOS_PER_MILLI, 0.001, false);
+    private final ExponentialMovingAverage appendsOutstanding = new ExponentialMovingAverage(20, 0.001, false);
```

Comment:
I selected the new values because my local testing showed they greatly improved the max latency. 
Pavel's tests confirmed this as they showed much lower tail variance in batch sizes, and with them reverted it doubled the p99 and tripled the p999. 
I think they are better as they are.